components/renku_data_services/base_models/core.py

```diff
@@ -207,6 +207,62 @@ def __repr__(self) -> str:
         return self.value
 
 
+@dataclass(frozen=True, eq=True)
+class EntityPath:
```

Comment:
suggestion: should there be some validation here that it only has 1 or 2 segments and not more? 

Or more generally, have subclasses for the different slugs that validate number of segments, and then the typechecker can make sure that you don't pass a dataset slug to a method that expects a namespace?

if this class (or a subclass) had a limit to 1 or 2, you could also add properties like `first` and `second` (terrible names, just to illustrate) and set `__match_args__ = ("first","second")` and then some of the code below could just be

```
path = EntityPath.from_string(body.namespace)
namespace, project = path # could just pattern match directly on the line above as well
```

But maybe that's overengineering, hence just a suggestion :man_shrugging:

Reply:
No that is a good idea. Because in a subsequent PR I use it in a match/case statement. So that is pretty much exactly what I need. I will add this.

Reply:
Done in https://github.com/SwissDataScienceCenter/renku-data-services/pull/617/commits/81f1b6ccdcc254da6825fb6b7729d068632f2259